### PR TITLE
Power go client update from 1.0.83 to 1.0.85

### PR DESCRIPTION
**Pull request by Madhan-SWE**
Fixes: https://github.com/ppc64le-cloud/powervs-csi-driver/issues/56 https://github.com/ppc64le-cloud/powervs-csi-driver/issues/42

### CODE REVIEW

File: pkg/cloud/powervs.go [hunk 1]
```diff
@@ -236,6 +237,7 @@ func (p *powerVSCloud) GetImageByID(imageID string) (*PVMImage, error) {
 func (p *powerVSCloud) CreateDisk(volumeName string, diskOptions *DiskOptions) (disk *Disk, err error) {
 	var volumeType string
 	capacityGiB := util.BytesToGiB(diskOptions.CapacityBytes)
+	diskSize := float64(capacityGiB)
```
Comment (mkumatag) [inline pkg/cloud/powervs.go:240]: can this be merged with the above line itself: `float64(util.BytesToGiB(diskOptions.CapacityBytes))`

File: pkg/cloud/powervs.go [hunk 2]
```diff
@@ -260,16 +269,16 @@ func (p *powerVSCloud) CreateDisk(volumeName string, diskOptions *DiskOptions) (
...
 		return false, err
 	}
 
 	return true, nil
 }
 
-func (p *powerVSCloud) AttachDisk(volumeID string, nodeID string) (err error) {
-	_, err = p.volClient.Attach(nodeID, volumeID, p.cloudInstanceID, TIMEOUT)
+func (p *powerVSCloud) AttachDisk(volumeName string, volumeID string, nodeID string) (err error) {
+	err = p.volClient.Attach(nodeID, volumeName)
```
Comment (mkumatag) [inline pkg/cloud/powervs.go:281]: I know its out of context but any idea why we need both id and name for the attach api?

cc @yussufsh

**Review by mkumatag — CHANGES_REQUESTED** [review]
few minor comments

File: pkg/cloud/powervs.go (hunk 2 above)
Comment (yussufsh) [reply pkg/cloud/powervs.go:281]: It's vm instance id 1st param and volume name or id for the 2nd.

Comment (mkumatag) [reply pkg/cloud/powervs.go:281]: thanks @yussufsh for the information, @Madhan-SWE  can we just use the volumeID to the `Attach` api, that way we don't need to our AttachDisk api

Comment (Madhan-SWE) [reply pkg/cloud/powervs.go:281]: `p.volClient.Attach` Requires volume Name as an argument.
Also in our AttachDisk method we use `WaitForVolumeState` that requires `volumeId` as an argument(it internally uses `volClient.Get(volumeID)`).

Hence, we need both volumeId and volumeName in our attach disk method.

Comment (Madhan-SWE) [reply pkg/cloud/powervs.go:281]: @yussufsh the second param for `Attach` could be wither volumeName or volumeId ?

File: pkg/cloud/powervs.go (hunk 1 above)
Comment (Madhan-SWE) [reply pkg/cloud/powervs.go]: fixed

File: pkg/cloud/powervs.go [hunk 3]
```diff
@@ -235,7 +236,7 @@ func (p *powerVSCloud) GetImageByID(imageID string) (*PVMImage, error) {
 
 func (p *powerVSCloud) CreateDisk(volumeName string, diskOptions *DiskOptions) (disk *Disk, err error) {
 	var volumeType string
-	capacityGiB := util.BytesToGiB(diskOptions.CapacityBytes)
+	capacityGiB := float64(util.BytesToGiB(diskOptions.CapacityBytes))
```
Comment (mkumatag) [inline pkg/cloud/powervs.go:239]: this looks to be a little odd, we are converting int64 => float64 and later in the end again back to int64, instead we can just change to float64 directly while assigning to the field, e.g:

```
	dataVolume := &models.CreateDataVolume{
 		Name:      &volumeName,
 		Size:      pointer.Float64Ptr(float64(capacityGiB)),
 		Shareable: &diskOptions.Shareable,
 		DiskType:  volumeType,
 	}
```
> Note: pointer module is from https://github.com/kubernetes/utils/blob/master/pointer/pointer.go

or change the `CapacityGiB` type to float64 if that is more convenient.

File: pkg/cloud/powervs.go [hunk 4]
```diff
@@ -171,14 +172,14 @@ func newPowerVSCloud(cloudInstanceID string, debug bool) (Cloud, error) {
...
+	piSession, err := ibmpisession.New(bxSess.Config.IAMAccessToken, region, debug, user.Account, zone)
 	if err != nil {
 		return nil, err
 	}
 
-	volClient := instance.NewIBMPIVolumeClient(piSession, cloudInstanceID)
+	volClient := instance.NewIBMPIVolumeClient(context.Background(), piSession, cloudInstanceID)
 	pvmInstancesClient := instance.NewIBMPIInstanceClient(context.Background(), piSession, cloudInstanceID)
-	imageClient := instance.NewIBMPIImageClient(piSession, cloudInstanceID)
+	imageClient := instance.NewIBMPIImageClient(context.Background(), piSession, cloudInstanceID)
```
Comment (mkumatag) [inline pkg/cloud/powervs.go:182]: Background can be created in one place in the beginning and pass same context to all the functions.

**Review by mkumatag — CHANGES_REQUESTED** [review]

**Review by mkumatag — APPROVED** [review]
/lgtm

File: pkg/cloud/powervs.go (hunk 3 above)
Comment (Madhan-SWE) [reply pkg/cloud/powervs.go]: fixed

File: pkg/cloud/powervs.go (hunk 4 above)
Comment (Madhan-SWE) [reply pkg/cloud/powervs.go]: fixed

**Comment by mkumatag** [comment]
@Madhan-SWE your branch is dirty again :P

**Comment by Madhan-SWE** [comment]
> @Madhan-SWE your branch is dirty again :P

Yes :-( 

Raised a new PR.

**Comment by Madhan-SWE** [comment]
Please review and approve this: https://github.com/ppc64le-cloud/powervs-csi-driver/pull/59

**Pull request closed without merging**